### DISCUSSION TITLE
Any chance of setting setRXBandwidth more flexible?

### DISCUSSION THREAD

**Discussion by tauesdospi** [jgromes/RadioLib#943] [Ideas]
Hi
I have noticed setRXBandwidth in SX126x modules is discrete and only accepts a set of values:
....
  if(fabs(rxBw - 4.8) <= 0.001) {
    this->rxBandwidth = RADIOLIB_SX126X_GFSK_RX_BW_4_8;
  } else if(fabs(rxBw - 5.8) <= 0.001) {
    this->rxBandwidth = RADIOLIB_SX126X_GFSK_RX_BW_5_8;
....

What do you think about relaxing this behavior and setting the rxBandwidth to the next upper step for any value between one step and another?

Eg.  setRxBandwidh(3) ==> equals to 4.8
setRxBandwidth(5) ==> equals to 5.8
etc..
also a function like "getRealRxBandwidth" to know the exact RxBandwidth value may help.

Would you consider this?
Regards
Antonio

**Comment by jgromes** [comment]
Actually I'm considering completely dropping those `float`s and have configuration via a defined macro or enum value. It's just easier to use. This will however break backwards compatibility, so I'm saving it to next major release of the library.
